content/articles/support-vector-regression-in-python/index.md

```diff
@@ -0,0 +1,303 @@
+
+```
+Let's simulteneously print and check if our two variiables were really scaled.
+```bash
+print("Scaled X_l:")
+print(X_l)
+print("Scaled y_p:")
+print(y_p)
+
+```
+Output:
+```bash
+Scaled X_l:
+[[-1.5666989 ]
+ [-1.21854359]
+ [-0.87038828]
+ [-0.52223297]
+ [-0.17407766]
+ [ 0.17407766]
+ [ 0.52223297]
+ [ 0.87038828]
+ [ 1.21854359]
+ [ 1.5666989 ]]
+Scaled y_p:
+[[-0.72004253]
+ [-0.70243757]
+ [-0.66722767]
+ [-0.59680786]
+ [-0.49117815]
+ [-0.35033854]
+ [-0.17428902]
+ [ 0.17781001]
+ [ 0.88200808]
+ [ 2.64250325]]
+
+```
+As we can see from the obtained output, both variables were scaled within the range -3 and +3.
+
+Our data is now ready to implement our SVR model. However, before we can do so, we will first visualize the data to know the nature of the SVR model that best fits it. So let us create a scatter plot of our two variables.
+
+```python
+plt.scatter(X_l, y_p, color = 'red') # plotting the training set
+plt.title('Scatter Plot') # adding a tittle to our plot
+plt.xlabel('Levels') # adds a label to the x-axis
+plt.ylabel('Salary') # adds a label to the y-axis
+plt.show() # prints
+
+```
+![Scatter plot](/engineering-education/support-vector-regression-in-python/plot0.png)
+
+The plot shows a nonlinear relationship between the `Levels` and the `Salary`. Therefore, we will use a kernelized version of the SVR model.
+
+### Implementing the Support Vector Regression
+
+To implement our model, first, we need to import it from the sk-learn and create an object to itself. Since we declared our data to be nonlinear, we will pass a kernel as `rbf`, which stands for, *Radial Basis function kernel*. After we have declared the kernel function, we will fit our data on the object. The following program performs these rules.
+
+```python
+# import the model
+from sklearn.svm import SVR
+ # create the model object
+regressor = SVR(kernel = 'rbf')
+# fit the model on the data
+regressor.fit(X_l, y_p)
+
+```
+Since the model is now ready, we can use it and make predictions. Let's see how we do so.
+```python
+A=regressor.predict(StdS_X.transform([[6.5]]))
+A
+
+```
+Output:
+
+```bash
+array([-0.27861589])
+
+```
+As we can see, the model prediction values are for the scaled study variable. The required value in the business is the output of the unscaled data. So we need to get back to the real scale of the study variable.
+
+To go back to the real study variable, we will write a program whose objective is to take the predicted values on the scaled range and transform them to the actual scale. To do so, we will take an inverse of the transformation we did on the study variable on our prediction function.
+
+We need to note that the predicted values are returned in 1D array. However, we had reshaped our study variable into 2D array. So, for any predicted value to correspond with such a new dimension, it must be transformed to 2D; otherwise, we get an error. So, let's implement these commands and get the required value.
```

Comment:
Why are we exactly transforming 1D to 2D array? It can be more descriptive here. It's not very clear.